File: app/src/main/java/com/doubean/ford/data/vo/GroupPostCategoryRule.java
package com.doubean.ford.data.vo;

import androidx.room.Entity;

import java.util.List;

@Entity
public class GroupPostCategoryRule {
    int id;
    List<Integer> groupIds;
    List<Integer> groupPostTagId;
    String regex;
    String notRegex;
    boolean strict;
}

File: app/src/main/java/com/doubean/ford/data/vo/GroupPostCategoryRule.kt
package com.doubean.ford.data.vo

import androidx.room.Entity

@Entity
class GroupPostCategoryRule {
    var id = 0
    var groupIds: List<Int>? = null
    var groupPostTagId: List<Int>? = null
    var regex: String? = null
    var notRegex: String? = null
    var strict = false
}